### PR TITLE
Support to build with JDK 11

### LINKED ISSUE #12578: Support JDK 11 build
### Describe the feature you would like.

I want to build shadingsphere with JDK 11

### PR DESCRIPTION
**Pull request by zhfeng**
Fixes #12578

Changes proposed in this pull request:
- Add dependency of ```javax.xml.bind::jaxb-api``` and ```javax.annotation::javax.annotation-api```
- minor fix the generic type complitation error in ```SchemaYamlSwapper:swapToYamlConfiguration```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -887,6 +890,27 @@
         </plugins>
     </build>
     
+    <profiles>
```
Comment (TeslaCN) [inline pom.xml:893]: Hi @zhfeng 
Could you fix the indent of the tags?

File: pom.xml [hunk 2]
```diff
@@ -887,6 +890,27 @@
...
+        <profile>
+            <id>jdk11+</id>
+            <activation>
+                <jdk>[11,)</jdk>
```
Comment (TeslaCN) [inline pom.xml:897]: The javax annotation was removed since JDK 9. So maybe the low bound should be 1.9?

File: pom.xml (hunk 1 above)
Comment (zhfeng) [reply pom.xml:893]: Done

File: pom.xml (hunk 2 above)
Comment (zhfeng) [reply pom.xml:897]: Since JDK 9 is not a LTS verion, I think it will not be in the supported JDK list.

**Review by TeslaCN — APPROVED** [review]
LGTM

**Comment by zhfeng** [comment]
Thanks @terrymanu and it looks like there are some failing tests when runnning with JDK 11. I'm investigating and will creat following PRs to fix them.

**Pull request merged**